msm/src/circuit_design/witness.rs

```diff
@@ -10,14 +10,24 @@ use crate::{
 use ark_ff::PrimeField;
 use kimchi::circuits::domains::EvaluationDomains;
 use log::debug;
-use std::{collections::BTreeMap, iter};
+use std::{collections::BTreeMap, iter, marker::PhantomData};
 
-/// Witness builder environment. Operates
-pub struct WitnessBuilderEnv<F: PrimeField, const CIX_COL_N: usize, LT: LookupTableID> {
+/// Witness builder environment. Operates on multiple rows at the same
+/// time. `CIx::N_COL` must be equal to `N_COL`; passing these two
```

Comment:
Cannot this be forced in Rust? Something similar to `CIx: ColumnIndexer<N_COL = N_COL>`. I do similar things when telling the traits the equalities that must hold between folding types and o1vm's.

Reply:
I already spent some time before trying to do it, but just tried again, and my conclusions are negative: it's not easy.
- https://docs.rs/type-equals/latest/type_equals/ doesn't seem to work for `const` equality
- https://docs.rs/type-equalities/latest/type_equalities/index.html seems too experimental
- https://stackoverflow.com/questions/60138397/how-to-test-for-type-equality-in-rust something unstable + something hacky

Reply:
Hope it were possible though...

Reply:
Awesome, thanks for trying though!